pkg/avancement/service_manager.go

```diff
@@ -131,7 +143,21 @@ func (s *ServiceManager) cloneRepo(repoURL, branch string) (git.Repo, error) {
 	}
 	repo, err := s.repoFactory(repoURL, path.Join(s.cacheDir, encode(repoURL, branch)), s.debug)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create repo for URL %s: %w", repoURL, err)
+		errorToDisplay := err
+		// This can contain the access token so omit it here too
+		asRealURL, parseErr := url.Parse(repoURL)
+		if parseErr != nil {
+			// Intentionally don't display the URL it tried to use here - yeah it'd be helpful but
+			// may contain sensitive data
+			return nil, errors.New("failed to create repository - could not parse URL")
+		}
+		asRealURL.User = nil // This is the important part, without it the test should fail
+		repoURLToDisplay := asRealURL
+		if strings.Contains(err.Error(), "status code 128") {
```

Comment:
This would become `if git.IsGitError(err) { ... }`

Reply:
Hmm, what's git.IsGitError? I did a grep, see we've got a `git` package but no such function there, or are you suggesting I make one and refactor a little?

Reply:
`git.IsGitError` comes from this

https://github.com/rhd-gitops-example/services/pull/28#discussion_r400982599